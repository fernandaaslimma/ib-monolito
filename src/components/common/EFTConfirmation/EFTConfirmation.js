import React, { Component } from "react";
import { Button, AlertMessage } from "react-bocombbm-components";
import { object, func } from "prop-types";
import { translate } from "../../../utils/i18n";
import Checkbox from "../../common/Checkbox2";
import Tag from "../../common/Tag";
import { formatCNPJ } from "../../../utils/formatNumber";

import {
  EFTModalWrapper,
  MainContainer,
  LeftContainer,
  CenterContainer,
  RightContainer,
  FooterContainer,
  UpperDiv,
  LowerDiv,
  SubDiv,
  Title,
  SubTitle,
  LargeSubTitle,
  WrapperSave,
  WrapperCheckbox,
  WrapperAlert,
  TitleSave
} from "./styles";

class EFTConfirmation extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoading: false,
      saveFavoredAccount: false,
      permissionToSaveAccount: false
    };
    this.handleConfirmButton = this.handleConfirmButton.bind(this);
    this.handleSaveFavoredAccount = this.handleSaveFavoredAccount.bind(this);
  }

  componentDidMount() {
    const { approvers, userInfo } = this.props;

    const userApprover = approvers.find(c => c.uid === userInfo.document);

    this.setState({
      permissionToSaveAccount:
        userApprover !== undefined
          ? userApprover.permissionToSaveAccount
          : false
    });
  }

  handleSaveFavoredAccount() {
    this.setState({
      saveFavoredAccount: this.state.saveFavoredAccount ? false : true
    });
  }

  handleConfirmButton() {
    this.setState({
      isLoading: true
    });
    this.props.onConfirm(this.state.saveFavoredAccount);
  }

  render() {
    const {
      transferData,
      favoredData,
      originAccount,
      approvers,
      closeModal,
      currentAccount,
      isNewAccount
    } = this.props;

    const {
      isLoading,
      saveFavoredAccount,
      permissionToSaveAccount
    } = this.state;

    return (
      <EFTModalWrapper>
        <MainContainer>
          <LeftContainer>
            <UpperDiv>
              <Title>{translate("DATE")}</Title>
              <SubTitle data-test="reviewTransferDate">
                {transferData.date}
              </SubTitle>
            </UpperDiv>
            <LowerDiv>
              <Title>{translate("VALUE")}</Title>
              <SubTitle data-test="reviewTransferAmount">
                {`${translate("CURRENCY_UNIT")} ${transferData.value}`}
              </SubTitle>
            </LowerDiv>
          </LeftContainer>

          <CenterContainer>
            <UpperDiv>
              <SubDiv>
                <Title>{translate("FAVORED")}</Title>
                <LargeSubTitle data-test="reviewRecipientName">
                  {currentAccount.name}
                </LargeSubTitle>
              </SubDiv>
              <SubDiv />
              <SubDiv>
                <Title>{translate("CNPJ")}</Title>
                <SubTitle data-test="reviewRecipientTaxId">
                  {formatCNPJ(favoredData.CNPJ)}
                </SubTitle>
              </SubDiv>
            </UpperDiv>
            <LowerDiv>
              <SubDiv width="46%">
                <Title>{translate("DESTINATION_BANK")}</Title>
                <SubTitle data-test="reviewRecipientBank">
                  {favoredData.bankCode} {favoredData.bank}
                </SubTitle>
              </SubDiv>
              <SubDiv width="20%">
                <Title>{translate("AGENCY")}</Title>
                <SubTitle data-test="reviewBankBranch">
                  {favoredData.agency}
                </SubTitle>
              </SubDiv>
              <SubDiv>
                <Title>{translate("ACCOUNT")}</Title>
                <SubTitle data-test="reviewRecipientBankAccount">
                  {favoredData.account}
                  {favoredData.verifyDigit && `-${favoredData.verifyDigit}`}
                </SubTitle>
              </SubDiv>
            </LowerDiv>
          </CenterContainer>

          <RightContainer>
            <UpperDiv>
              <Title>{translate("ORIGIN_ACCOUNT")}</Title>
              <SubTitle data-test="reviewOriginAccount">
                {originAccount.number}
              </SubTitle>
            </UpperDiv>
            <LowerDiv>
              <Title>{translate("APPROVERS")}</Title>
              <SubTitle>
                {approvers.map((c, i) => {
                  if (i === approvers.length - 1) {
                    return `${c.name}.`;
                  }
                  return `${c.name}, `;
                })}
              </SubTitle>
            </LowerDiv>
          </RightContainer>
        </MainContainer>

        {permissionToSaveAccount && (
          <WrapperSave>
            <TitleSave>
              {translate("EFT_SAVE_BANK_ACCOUNT")}
              <Tag title={translate("NEW")} color={"#D5E3F8"} />
            </TitleSave>

            <WrapperCheckbox>
              <Checkbox
                type="switch"
                name="switchBox"
                dataTest="switchSaveAccount"
                checked={saveFavoredAccount}
                onChange={this.handleSaveFavoredAccount}
              />
            </WrapperCheckbox>
          </WrapperSave>
        )}
        {!isNewAccount && !permissionToSaveAccount && (
          <WrapperAlert>
            <AlertMessage icon="Attention" type="neutral">
              {translate("EFT_ACCOUNTS_SAVE_DISCLAIMER")}
            </AlertMessage>
          </WrapperAlert>
        )}
        <FooterContainer>
          <Button
            type="outline"
            dataTest="Cancel"
            onClick={() => closeModal()}
            disabled={!!isLoading}
          >
            {translate("CANCEL_BTN")}
          </Button>

          <Button
            type="conclusive"
            dataTest="Confirm"
            onClick={this.handleConfirmButton}
            disabled={!!isLoading}
            loading={isLoading}
          >
            {translate("CONFIRM_BTN")}
          </Button>
        </FooterContainer>
      </EFTModalWrapper>
    );
  }
}

EFTConfirmation.defaultProps = {
  transferData: null,
  favoredData: null,
  originAccount: null
};

EFTConfirmation.propTypes = {
  transferData: object,
  favoredData: object,
  originAccount: object,
  closeCallback: func
};

export default EFTConfirmation;
