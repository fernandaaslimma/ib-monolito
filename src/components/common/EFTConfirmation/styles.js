import styled from "styled-components";
import { rem, remFontSize } from "../../../styles/tools";
import { lightgrey, grey90, gray300 } from "../../../styles/settings";
import { LinkTag } from "../../../styles/objects/actionElements";

export const EFTModalWrapper = styled.div`
  border-top: solid ${rem(1)} ${lightgrey};

  ${LinkTag} {
    margin-top: ${rem(20)};
    height: ${rem(40)};
    line-height: ${rem(40)};
    width: ${rem(120)};
  }
`;

export const MainContainer = styled.div`
  padding: ${rem(30)} ${rem(20)} ${rem(30)};
  display: flex;
`;

export const LeftContainer = styled.div`
  border: solid ${rem(1)} ${lightgrey};
  border-radius: ${rem(5)};
  padding: ${rem(20)} ${rem(25)} ${rem(20)};
  margin-left: ${rem(10)};
  margin-right: ${rem(10)};
  width: 18%;
  display: inline-block;
`;

export const CenterContainer = styled.div`
  border: solid ${rem(1)} ${lightgrey};
  border-radius: ${rem(5)};
  padding: ${rem(20)} ${rem(25)} ${rem(20)};
  margin-left: ${rem(10)};
  margin-right: ${rem(10)};
  width: 51%;
  display: inline-block;
`;

export const RightContainer = styled.div`
  border: solid ${rem(1)} ${lightgrey};
  border-radius: ${rem(5)};
  padding: ${rem(20)} ${rem(25)} ${rem(20)};
  margin-left: ${rem(10)};
  margin-right: ${rem(10)};
  width: 26%;
  display: inline-block;
`;

export const FooterContainer = styled.div`
  display: inline-flex;
`;

export const UpperDiv = styled.div`
  height: ${rem(65)};
`;

export const LowerDiv = styled.div`
  height: ${rem(55)};
`;

export const SubDiv = styled.div`
  width: ${({ width = false }) => (!width ? "33%" : width)};
  height: 100%;
  float: left;
`;

export const Title = styled.h1`
  color: ${grey90};
  font-family: Lato;
  font-size: ${remFontSize(10)};
  font-weight: bold;
  letter-spacing: ${rem(0.4)};
  text-transform: uppercase;
  text-align: left;
`;

export const SubTitle = styled.h1`
  color: ${grey90};
  font-family: Lato;
  font-size: ${remFontSize(13.3)};
  letter-spacing: ${rem(0.4)};
  margin-top: ${rem(8)};
  text-align: left;
`;

export const LargeSubTitle = styled.h1`
  color: ${grey90};
  font-family: Lato;
  font-size: ${remFontSize(13.3)};
  letter-spacing: ${rem(0.4)};
  margin-top: ${rem(8)};
  text-align: left;
  width: 180%;
`;

export const WrapperSave = styled.div`
  display: grid;
  grid-template-columns: 56% auto;
  text-align: end;
  margin-bottom: 20px;
`;

export const WrapperCheckbox = styled.div`
  margin: 3px 0 0 23px;
`;

export const WrapperAlert = styled.div`
  display: flex;
  justify-content: center;
  margin-bottom: ${rem(32)};
`;

export const TitleSave = styled.div`
  font-family: Lato;
  font-style: normal;
  font-weight: 500;
  font-size: ${rem(14)};
  line-height: ${rem(17)};
  color: ${gray300};
  display: flex;
  justify-content: flex-end;

  span {
    margin-left: ${rem(8)};
  }
`;
