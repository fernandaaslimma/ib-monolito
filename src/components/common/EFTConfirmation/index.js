import EFTConfirmation from "./EFTConfirmation";

import { connect } from "redux-zero/react";
import { combineActions } from "redux-zero/utils";
import modalActions from "../../../actions/modal";

export default connect(
  null,
  combineActions(modalActions)
)(EFTConfirmation);
