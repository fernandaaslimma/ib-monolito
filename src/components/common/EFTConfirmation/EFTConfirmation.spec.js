import React from "react";
import { shallow, mount } from "enzyme";
import EFTConfirmation from "./EFTConfirmation";

const props = {
  userInfo: {
    givenName: "Yuri Ramos"
  },
  closeModal: jest.fn(),
  onConfirm: jest.fn(),
  transferData: {
    date: "01/01/2019",
    value: "2321312"
  },
  favoredData: {
    favored: "GTL Logística de transporte",
    CNPJ: "59.272.491/0001-72",
    bank: "Itaú Unibanco",
    agency: "8230-2",
    account: "19323-2",
    verifyDigit: "0"
  },
  originAccount: {
    number: "106 2 300431-1"
  },
  approvers: [
    {
      name: "Yuri Ramos",
      permissionToSaveAccount: true
    },
    { name: "André Leitão", permissionToSaveAccount: true }
  ],
  currentAccount: {
    name: "Yuri",
    document: "11111111111"
  }
};

describe("EFTConfirmation component", () => {
  it("should match snapshot", () => {
    expect(shallow(<EFTConfirmation {...props} />)).toMatchSnapshot();
  });

  it('should render an "EFTConfirmation" tag', () => {
    expect(shallow(<EFTConfirmation {...props} />)).toMatchSnapshot();
    expect(
      mount(<EFTConfirmation {...props} />).find("EFTConfirmation").length
    ).toBe(1);
  });
});

describe("click simulation tests", () => {
  it("should close modal properly on confirm", () => {
    jest.spyOn(EFTConfirmation.prototype, "handleConfirmButton");
    const wrapper = shallow(<EFTConfirmation {...props} />);
    const confirmButton = wrapper.find('Button[dataTest="Confirm"]');
    confirmButton.simulate("click");

    expect(wrapper.state()).toEqual({
      isLoading: true,
      permissionToSaveAccount: true,
      saveFavoredAccount: false
    });

    expect(EFTConfirmation.prototype.handleConfirmButton.calledOnce);
    expect(props.onConfirm).toHaveBeenCalled();
  });
});
