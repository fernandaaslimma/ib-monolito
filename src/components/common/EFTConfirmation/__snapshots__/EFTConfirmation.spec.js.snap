// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EFTConfirmation component should match snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <styled.div>
        <styled.h1 />
        <styled.h1
          data-test="reviewTransferDate"
        >
          01/01/2019
        </styled.h1>
      </styled.div>
      <styled.div>
        <styled.h1 />
        <styled.h1
          data-test="reviewTransferAmount"
        >
          undefined 2321312
        </styled.h1>
      </styled.div>
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.div>
          <styled.h1 />
          <styled.h1
            data-test="reviewRecipientName"
          >
            Yuri
          </styled.h1>
        </styled.div>
        <styled.div />
        <styled.div>
          <styled.h1 />
          <styled.h1
            data-test="reviewRecipientTaxId"
          >
            59.272.491/0001-72
          </styled.h1>
        </styled.div>
      </styled.div>
      <styled.div>
        <styled.div
          width="46%"
        >
          <styled.h1 />
          <styled.h1
            data-test="reviewRecipientBank"
          >
             
            Itaú Unibanco
          </styled.h1>
        </styled.div>
        <styled.div
          width="20%"
        >
          <styled.h1 />
          <styled.h1
            data-test="reviewBankBranch"
          >
            8230-2
          </styled.h1>
        </styled.div>
        <styled.div>
          <styled.h1 />
          <styled.h1
            data-test="reviewRecipientBankAccount"
          >
            19323-2
            -0
          </styled.h1>
        </styled.div>
      </styled.div>
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.h1 />
        <styled.h1
          data-test="reviewOriginAccount"
        >
          106 2 300431-1
        </styled.h1>
      </styled.div>
      <styled.div>
        <styled.h1 />
        <styled.h1>
          Yuri Ramos, 
          André Leitão.
        </styled.h1>
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <Tag
        color="#D5E3F8"
      />
    </styled.div>
    <styled.div>
      <Checkbox
        checked={false}
        dataTest="switchSaveAccount"
        name="switchBox"
        onChange={[Function]}
        type="switch"
      />
    </styled.div>
  </styled.div>
  <styled.div>
    <Button
      dataTest="Cancel"
      disabled={false}
      onClick={[Function]}
      spacing={
        Object {
          "bottom": "xxs",
          "left": "xxs",
          "right": "xxs",
          "top": "xxs",
        }
      }
      type="outline"
    />
    <Button
      dataTest="Confirm"
      disabled={false}
      loading={false}
      onClick={[Function]}
      spacing={
        Object {
          "bottom": "xxs",
          "left": "xxs",
          "right": "xxs",
          "top": "xxs",
        }
      }
      type="conclusive"
    />
  </styled.div>
</styled.div>
`;

exports[`EFTConfirmation component should render an "EFTConfirmation" tag 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <styled.div>
        <styled.h1 />
        <styled.h1
          data-test="reviewTransferDate"
        >
          01/01/2019
        </styled.h1>
      </styled.div>
      <styled.div>
        <styled.h1 />
        <styled.h1
          data-test="reviewTransferAmount"
        >
          undefined 2321312
        </styled.h1>
      </styled.div>
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.div>
          <styled.h1 />
          <styled.h1
            data-test="reviewRecipientName"
          >
            Yuri
          </styled.h1>
        </styled.div>
        <styled.div />
        <styled.div>
          <styled.h1 />
          <styled.h1
            data-test="reviewRecipientTaxId"
          >
            59.272.491/0001-72
          </styled.h1>
        </styled.div>
      </styled.div>
      <styled.div>
        <styled.div
          width="46%"
        >
          <styled.h1 />
          <styled.h1
            data-test="reviewRecipientBank"
          >
             
            Itaú Unibanco
          </styled.h1>
        </styled.div>
        <styled.div
          width="20%"
        >
          <styled.h1 />
          <styled.h1
            data-test="reviewBankBranch"
          >
            8230-2
          </styled.h1>
        </styled.div>
        <styled.div>
          <styled.h1 />
          <styled.h1
            data-test="reviewRecipientBankAccount"
          >
            19323-2
            -0
          </styled.h1>
        </styled.div>
      </styled.div>
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.h1 />
        <styled.h1
          data-test="reviewOriginAccount"
        >
          106 2 300431-1
        </styled.h1>
      </styled.div>
      <styled.div>
        <styled.h1 />
        <styled.h1>
          Yuri Ramos, 
          André Leitão.
        </styled.h1>
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <Tag
        color="#D5E3F8"
      />
    </styled.div>
    <styled.div>
      <Checkbox
        checked={false}
        dataTest="switchSaveAccount"
        name="switchBox"
        onChange={[Function]}
        type="switch"
      />
    </styled.div>
  </styled.div>
  <styled.div>
    <Button
      dataTest="Cancel"
      disabled={false}
      onClick={[Function]}
      spacing={
        Object {
          "bottom": "xxs",
          "left": "xxs",
          "right": "xxs",
          "top": "xxs",
        }
      }
      type="outline"
    />
    <Button
      dataTest="Confirm"
      disabled={false}
      loading={false}
      onClick={[Function]}
      spacing={
        Object {
          "bottom": "xxs",
          "left": "xxs",
          "right": "xxs",
          "top": "xxs",
        }
      }
      type="conclusive"
    />
  </styled.div>
</styled.div>
`;
